src/openfermion/utils/_trotter_qasm_conversion.py

```diff
@@ -0,0 +1,331 @@
+		raise ValueError("Invalid number of trotter steps: " + str(trotter_steps))
+		
+	if not isinstance(hamiltonian, QubitOperator):
+		raise TypeError("Hamiltonian must be a QubitOperator.")
+	if not hamiltonian.terms:
+		raise TypeError("Hamiltonian must be a non-empty QubitOperator.")
+	if not isinstance(trotter_steps, int):
+		raise TypeError("trotter_steps must be an int")
+	if not isinstance(trotter_order, int):
+		raise TypeError("trotter_order must be an int")
+	if not isinstance(k_exp, float):
+		raise TypeError("k_exp must be a float")
+		
+		
+	num_qubits = 0
+	for term in hamiltonian.terms.items():
+		for item in term[0]:
+			if item[0] > num_qubits:
+				num_qubits = item[0]
+	num_qubits += 1
+	ret_val = []
+	
+	if len(term_ordering) == 0:
+		for term in hamiltonian.terms.items():
+			term_ordering.append(term[0])
+	
+	if trotter_order == 1:
+		for step in range(trotter_steps):
+			for op in term_ordering:
+				ret_val.append(QubitOperator(op,hamiltonian.terms[op]) 
+					* k_exp/trotter_steps)
+
+	elif trotter_order == 2:
+		if len(term_ordering) < 2:
+			raise ValueError("Not enough terms in the Hamiltonian to do "+\
+				"second order trotterization")
+		for op in term_ordering[:-1]:
+			ret_val.append(QubitOperator(op,hamiltonian.terms[op]) * k_exp 
+				/ (2.0 * trotter_steps))
+		ret_val.append(QubitOperator(term_ordering[-1],
+			hamiltonian.terms[term_ordering[-1]] * k_exp / trotter_steps))
+
+		for op in reversed(term_ordering[:-1]):
+			ret_val.append(QubitOperator(op,hamiltonian.terms[op]) * k_exp 
+				/ (2.0 * trotter_steps))
+
+		for step in range(trotter_steps - 1):
+			ret_val.extend(ret_val)
+
+	elif trotter_order == 3:
+		if len(term_ordering) < 2:
+			raise ValueError("Not enough terms in the Hamiltonian to do "+\
+				"third order trotterization")
+		hamiltonian /= trotter_steps
+		hamiltonian *= k_exp
+		atemp = QubitOperator(term_ordering[0],hamiltonian.terms[term_ordering[0]])
+		btemp = term_ordering[1:]
+		third_order_trotter_helper(hamiltonian.terms, ret_val,atemp,btemp)
+
+		for step in range(trotter_steps - 1):
+			ret_val.extend(ret_val)
+	return (ret_val,num_qubits)
+
+
+
+def print_qubit_op_to_qasm(ostrm,qubop):
+	'''
+	This function exponentiates a single-Pauli-string QubitOperator
+	and prints it in QASM format to a supplied output stream
+
+	Params:
+		ostrm: file output stream
+		qubop: single Pauli-term QubitOperator to be printed
+		kexp (optional): float to be multiplied to the coefficient 
+	
+	Returns:
+		None
+	'''
+
+
+	for term in qubop.terms:
+
+		termCoeff = qubop.terms[term]
+
+		# List of operators and list of qubit ids
+		ops = list()
+		qids = list()
+		strBas1 = ""  # Basis rotations 1
+		strBas2 = ""  # Basis rotations 2
+		nl = "\n"  # New line
+
+		for p in term:	# p = single pauli term
+			qid = p[0]	# Qubit index
+			pop = p[1]	# Pauli op
+
+			qids.append(qid)  # Qubit index
+			ops.append(pop)	 # Pauli operator
+
+			if pop == 'X':
+				strBas1 += "H " + str(qid) + nl	 # Hadamard
```

Comment:
Don't append to strings in long loops, but rather append to a list then use
`'\n'.join(str_list)`
at the end to produce the final result.  The reason for this is that strings are immutable, so whenever you += them, it makes a copy of the whole thing, appends a bit, and reassigns.  If your circuits get long (i.e. 10^9 gates), this can be very slow.